sys-devel/llvm/llvm-3.8.1-r1.ebuild

```diff
@@ -169,6 +169,12 @@ src_prepare() {
 	# https://bugs.gentoo.org/show_bug.cgi?id=578392
 	eapply "${FILESDIR}"/llvm-3.8-soversion.patch
 
+	# support building llvm against musl-libc
+	use elibc_musl && eapply "${FILESDIR}"/llvm-3.8-musl-fixes.patch
+
+	# support "musl" as a valid environment type in llvm
+	eapply "${FILESDIR}"/llvm-3.8-musl-support.patch
```

Comment:
It is better to add patches unconditionally the way you do here.  If possible, on line 173, change the patch so its unconditional and see if you can get the build system of llvm to detect that you're on a musl system and act accordingly.